sfge/src/main/java/com/salesforce/rules/unusedmethod/RuleStateTracker.java

```diff
@@ -30,10 +28,32 @@ public class RuleStateTracker {
      */
     private final Set<MethodVertex> unusedMethods;
     /**
-     * A map used to cache every method call expression in a given class. Minimizing redundant
+     * A Set used to track every DefiningType for which we've cached values. Minimizing redundant
      * queries is a very high priority for this rule.
      */
-    private final Map<String, List<InvocableWithParametersVertex>> methodCallsByDefiningType;
```

Comment:
Splitting this into three maps that each track a separate implementation of `InvocableWithParametersVertex` makes this cleaner, since validators generally aren't interested in more than one or two types.